pkg/imgpkg/imagedesc/types.go

```diff
@@ -133,3 +140,23 @@ func (t ImageOrIndex) Tag() string {
 		panic("Unknown item")
 	}
 }
+
+func (t ImageOrIndex) MountableImage(registry RegistryRemoteImage) (regv1.Image, error) {
```

Comment:
i dont believe this pkg interacts with anything that talks to remote registries. let's move this method into ImageSet.

Reply:
This is interesting because this was not here in the first version of the PR and when I talked with @DennisDenuto we moved it here.
This package does talk with the registry to retrieve the descriptors for the images.

Current ImageSet knows how to Export and Import images but Mountable Images feel like more an implementation detail that ImageSet shouldn't care about because it just provides the images to go-container registry for upload. My initial thought was that the image should be able to determine if it is mountable or not. Unfortunately we can only mount images that are in the same registry and if you call `imgpkg` in a air gapped environment go-containerregistry will try to mount the image from a place that might not be accessible.

Another option that eventually we could think about is to push this concern to the Registry itself. Doing that could also be interesting but I am not 100% sure if we would have all the needed information when we are performing the writing. (Do not remember the code right now but I see no reason why we would not have all the information we need at that level)

Reply:
also this function is only applicable to images so if it takes ImageWithRef we wouldnt need first conditional. i also think that we can use imageBlobsCanBeMounted inside here so that question of mountability is encasulated in this function.

Reply:
>  Unfortunately we can only mount images that are in the same registry

that's what conceptually it means to be mountable so we model based on that. only place that has source and destination is ImageSet class we got going on. this function in current form does nothing related to mounting with the exception of being named Mountable. it could be named FromRegistry for example imho and it would reflect more accurately what it's doing. but does it really make sense to have a method that fetches image from a registry on an image itself (ImageWithRef is a regv1.Image)? imho not, hence proposal to move it to be on ImageSet as a prive method.

Reply:
@cppforlife and I moved MountableImage to ImageSet. A strong reason being that there isn't a concept of an ImageIndex being mountable.